types/react-native/index.d.ts

```diff
@@ -7973,6 +7974,29 @@ export class StatusBar extends React.Component<StatusBarProps> {
      * @param translucent Set as translucent.
      */
     static setTranslucent: (translucent: boolean) => void;
+
+    /**
+     * Push a StatusBar entry onto the stack.
+     * The return value should be passed to `popStackEntry` when complete.
+     *
+     * @param props Object containing the StatusBar props to use in the stack entry.
+     */
+    static pushStackEntry: (props: StatusBarProps) => any;
```

Comment:
What type of object can an entry be? And can you please add some test coverage that shows the usage of these APIs?

Reply:
`entry` is basically of type `StatusBarProps`, but if I'd change it to that, the following line would yield a `ts` error:
```javascript
useEffect(() => {
  statusBarStackEntry.current = StatusBar.pushStackEntry({
    barStyle: 'light-content',
    animated: true,
  });

  return () => StatusBar.popStackEntry(statusBarStackEntry.current);
}, []);
```
That's because React refs' `current` property is of type `React.MutableRefObject<null>.current: null`, and therefore TS reports the following errors
* `Type 'StatusBarProps' is not assignable to type 'null'. ts(2322)`
* `Argument of type 'null' is not assignable to parameter of type 'StatusBarProps'. ts(2345)`

for `pushStackEntry()` and `popStackEntry(statusBarStackEntry.current)`, respectively.
I honestly have no idea, how to fix that.

Regarding the tests: Since the `StatusBarTest` is extremely rudimentary anyway, I thought it would be no problem.

Reply:
The reason I asked for tests was mostly for _if_ there was the possibility for us to make the typings more explicit than those `any`s

Reply:
I added the `StatusBarProps` as type for `entry` and return type of `push` and `replace`